src/extra/recipes/cookiebox.py

```diff
@@ -1019,95 +984,148 @@ def plot_fit(self, tof_id: int):
         plt.title(f"TOF {tof_id}")
         plt.show()
 
-    def apply(self, run: DataCollection) -> xr.Dataset:
+    def load_trace(self, run: DataCollection, **extra_kwargs_adq: Dict[str, Any]) -> xr.Dataset:
```

Comment:
I'm not entirely sold on the idea of basically wrapping a constructor :thinking: Is it feasible to allow passing in a `DataArray` or `SourceData` object for the trace instead, in case one already has those lying around in a script? Or is it important that that e.g. a `DataArray` with the pulse information was constructed from an `AdqRawChannel` with specific settings?

Reply:
It is *extremely* important that the `AdqRawChannel` used for the `calibrate` step has exactly the same settings (with very few exceptions). This is why I split it in `load_trace` and `calibrate`, so that people can do it themselves and then call `calibrate` only, but at their own peril. Still, in most cases you would not want to do it yourself and you would want to use the same settings. I also don't like that I have to call the constructor myself, but given the constraints, are there alternatives?

Reply:
Ok, if it's that important then I think it's unavoidable. I also can't think of a better way.